crates/fluvio/src/producer/accumulator.rs

```diff
@@ -188,6 +189,7 @@ impl ProducerBatch {
                 Compression::None => 1.0,
                 Compression::Gzip | Compression::Snappy | Compression::Lz4 => 0.5,
             }) as usize
+            + Batch::<RawRecords>::default().write_size(0)
```

Comment:
Previosly, this computation only considered record size, but when we use a batch_size large enough like 1048588, when we added other Batch fields, the size was higher than that, therefore SPU was rejecting that with BatchTooBig, that is the reason for this change